scripts/ci/_wait-for.sh

```diff
@@ -21,21 +21,33 @@ USAGE
   exit "$exitcode"
 }
 
+check_availability() {
+  nc -z "$HOST" "$PORT" > /dev/null 2>&1
+
+  result=$?
+  if [ $result -eq 0 ] ; then
+    if [ $# -gt 0 ] ; then
+      exec "$@"
+    fi
+    exit 0
+  fi
+  sleep 1
+}
+
 wait_for() {
   echo "Waiting for $HOST on $PORT to be ready"
-  for i in `seq $TIMEOUT` ; do
-    nc -z "$HOST" "$PORT" > /dev/null 2>&1
+  
+  if [ $TIMEOUT -gt 0 ] ; then
+    for i in `seq $TIMEOUT` ; do
+      check_availability "$@"
+    done
+    echo "Operation timed out" >&2
+  else
+    while true ; do
```

Comment:
I don't particularly like the idea of possibly waiting for something forever in case the server doesn't come up. Can we come up with a reasonable default `TIMEOUT` instead? An hour?

    TIMEOUT=${TIMEOUT:-3600}

Reply:
If you do that you can get rid of the if/else and pull the function back into the business logic, so the exit logic is more readable.

Reply:
Undid all changes to `_wait-for.sh`